generator/overrides/gen.go

```diff
@@ -0,0 +1,435 @@
+					})
+				}
+
+				fieldChange := fieldChanges[cursor.Node().Pos()]
+				field := fieldChange.FieldInfo
+
+				overridesMarker := fieldChange.OverrideMarker
+
+				shouldSkip := func(overridesMarker FieldOverridesInclude) bool {
+					if overridesMarker.Omit ||
+						(overridesMarker.OmitInPlugin && g.IsForPluginOverrides) {
+						return true
+					}
+					return false
+				}
+
+				if shouldSkip(overridesMarker) {
+					cursor.Delete()
+					return true
+				}
+
+				if overridesMarker.Description != "" {
+					astField.Doc = updateComments(
+						astField, astField.Doc,
+						` *\+[^ ][^ ]*.*`,
+						` *\+`+overridesFieldMarker.Name+`.*`,
+						overridesMarker.Description,
+						"Overriding is done according to K8S strategic merge patch standard rules.",
+					)
+				}
+
+				if field.Name != newTypeToProcess.MandatoryKey {
+					// Make the field optional by default, unless typeToProcess contains a MandatoryKey nonempty field
+					jsonTag := field.Tag.Get("json")
+					if jsonTag != "" &&
+						jsonTag != "-" &&
+						!strings.Contains(jsonTag, ",inline") &&
+						!strings.Contains(jsonTag, ",omitempty") {
+						newJSONTag := jsonTag + ",omitempty"
+						astField.Tag.Value = strings.Replace(astField.Tag.Value, `json:"`+jsonTag+`"`, `json:"`+newJSONTag+`"`, 1)
+						astField.Doc = updateComments(
+							astField, astField.Doc,
+							`.*`,
+							` *`+regexp.QuoteMeta("+optional")+`.*`,
+							` +optional`,
+						)
+
+					}
+				}
+
+				processFieldType := func(ident *ast.Ident) *typeToProcess {
+					typeToOverride, existsInPackage := g.packageTypes[ident.Name]
+					if !existsInPackage {
+						return nil
+					}
+					ident.Name = ident.Name + g.suffix
+					return &typeToProcess{
+						OverrideTypeName: ident.Name,
+						TypeInfo:         typeToOverride,
+					}
+				}
+
+				var fieldTypeToProcess *typeToProcess
+
+				switch fieldType := astField.Type.(type) {
+				case *ast.ArrayType:
+					switch elementType := fieldType.Elt.(type) {
+					case *ast.Ident:
+						fieldTypeToProcess = processFieldType(elementType)
+						if fieldTypeToProcess != nil {
+							fieldTypeToProcess.MandatoryKey = strings.Title(genutils.GetPatchMergeKey(&field))
+						}
+					}
+				case *ast.Ident:
+					fieldTypeToProcess = processFieldType(fieldType)
+					if field.Markers.Get(genutils.UnionDiscriminatorMarker.Name) != nil {
+						enumValues := []string{}
+						for _, f := range typeToOverride.Fields {
+							pos := f.RawField.Pos()
+							fieldChange := fieldChanges[f.RawField.Pos()]
+							if pos != cursor.Node().Pos() &&
+								!shouldSkip(fieldChange.OverrideMarker) {
+								enumValues = append(enumValues, fieldChange.FieldInfo.Name)
+							}
+						}
+						kubebuilderAnnotation := "+kubebuilder:validation:Enum=" + strings.Join(enumValues, ";")
+						astField.Doc = updateComments(astField, astField.Doc, `.*`, ` *`+regexp.QuoteMeta("+kubebuilder:validation:Enum=")+`.*`, kubebuilderAnnotation)
```

Comment:
Any idea why the `+kubebuilder:validation:Enum` tags end up at the top of comment blocks? I can't quite figure it out.

Reply:
Because it is removed by the drop regexp (in case it exists but is wrong), and is added again as an additional line.
And as you guessed, `updateComments` prepends additional lines.